### ISSUE TITLE
Provide an include configuration mechanism 

### ISSUE THREAD

**Issue by arthurlogilab** [netdata/netdata#2360]
It would great if one could modify the configuration of netdata without touching the configuration shipped by the distribution. That way, when upgrades come along, those files are seen as not modified by the upgrade process and you get smoother upgrades, no diffs to read and new options as a bonus.

I don't know if there is a formalisation of this, but the include.d mechanism is great for that. That way we can edit (or ship via salt/ansible/puppet/chef/etc.) atomic files for atomic bits of configuration that don't interfere with the default configuration. 

Netdata already uses that for python.d health.d charts.d etc. It would be awesome (or is already the case ?) to be able to have the same for the other configurations: 

* netdata.conf -> add netdata.d/*.conf
* stream.conf -> add stream.d/*.conf
* fping.conf -> add fping.d/*.conf
* etc.

**Comment by ktsaou** [comment]
Hi,

For sure netdata configuration files have grown quite a bit.

well there are a few requests about `/etc/netdata`:

- #1126
- #1646
- #2009

There was also a request (I can't find now) about shipping netdata configurations somewhere in `/usr` and allow the users to overwrite any of them in `/etc/netdata`.

For sure this requires some more thinking.

For example, it would be great if packages could install/add jobs to specific modules (eg you install a web server and it also installs a file describing a new job for the `web_log` module of `python.d.plugin`).

**Comment by paulfantom** [comment]
Closing as a partial duplicate of #4182.

**Comment by ktsaou** [comment]
@paulfantom I think we should revisit this when #4182 is implemented. I don't want #4182 to include this one too, because it will too complex.

So, the idea of this, is to allow splitting config files in multiple parts and add an `include` directive on all kinds of files. I understand this gives quite some flexibility to users willing to control via their CI pipelines parts of netdata configuration.

My suggestion is to re-open this and do it after #4182.

**Comment by ktsaou** [comment]
@l2isbad 	the request we were talking before, about supporting directories of configuration files for plugin module jobs.

**Comment by ktsaou** [comment]
The motivation and a more generic solution is discussed in #4731

**Comment by cakrit** [comment]
Closing in favor of #6432
